lib/axon/loop.ex

```diff
@@ -536,7 +536,7 @@ defmodule Axon.Loop do
     # Build loss now so we can use it as a metric
     loss_fn = build_loss_fn(loss)
     {init_fn, step_fn} = train_step(model, loss_fn, optimizer)
-    output_transform = fn state -> state.step_state[:model_state] end
+    output_transform = fn state -> {state, state.step_state[:model_state]} end
```

Comment:
What I have in mind is keeping this the same `fn state -> state.step_state[:model_state] end`

And rather in `run` returning:

```
{state, output_transform.(state)}
```

Does that make sense?

Reply:
Yes, that does make sense. I've made the changes here https://github.com/elixir-nx/axon/pull/382/commits/bd927643760ccf54cec5e177d4b367d8e26aed1b . Should specific tests be added? I've just updated the current ones.